src/d3a/models/state.py

```diff
@@ -237,6 +274,18 @@ class PVState(ProductionState):
     Completely inherits ProductionState, but we keep this class for backward compatibility.
     """
 
+    def _calculate_unsettled_energy_kWh(self, measured_energy_kWh, time_slot):
```

Comment:
```suggestion
    def _calculate_unsettled_energy_kWh(self, measured_energy_kWh: float, time_slot: DateTime) -> float:
```

Reply:
Don't worry I was not blaming Faizan :D I was mostly thinking out loud here, and as long as it did not introduce circular imports then it is easy to add, and actually adapted it already. So all good